src/main/java/com/ldtteam/blockout/Loader.java

```diff
@@ -29,92 +30,90 @@
 {
     private static final DocumentBuilderFactory DOCUMENT_BUILDER_FACTORY = DocumentBuilderFactory.newInstance();
 
-    private static final Map<String, Constructor<? extends Pane>> paneConstructorMap = new HashMap<>();
+    private static final Map<ResourceLocation, Function<PaneParams,? extends Pane>> paneFactories = new HashMap<>();
     static
     {
-        register("view", View.class);
-        register("group", Group.class);
-        register("scrollgroup", ScrollingGroup.class);
-        register("list", ScrollingList.class);
-        register("text", Text.class);
-        register("button", ButtonVanilla.class);
-        register("buttonimage", ButtonImage.class);
-        register("label", Text.class); // TODO: remove, but we don't want to deal with xml changes now
-        register("input", TextFieldVanilla.class);
-        register("image", Image.class);
-        register("imagerepeat", ImageRepeatable.class);
-        register("box", Box.class);
-        register("itemicon", ItemIcon.class);
-        register("switch", SwitchView.class);
-        register("dropdown", DropDownList.class);
-        register("overlay", OverlayView.class);
-        register("gradient", Gradient.class);
-        register("zoomdragview", ZoomDragView.class);
-        register("treeview", TreeView.class);
+        register("view", View::new);
+        register("group", Group::new);
+        register("scrollgroup", ScrollingGroup::new);
+        register("list", ScrollingList::new);
+        register("text", Text::new);
+        register("button", Button::construct);
+        register("buttonimage", Button::construct); // TODO: remove, but we don't want to deal with xml changes now
+        register("toggle", ToggleButton::new);
+        register("label", Text::new); // TODO: remove, but we don't want to deal with xml changes now
+        register("input", TextFieldVanilla::new);
+        register("image", Image::new);
+        register("imagerepeat", ImageRepeatable::new);
+        register("box", Box::new);
+        register("itemicon", ItemIcon::new);
+        register("switch", SwitchView::new);
+        register("dropdown", DropDownList::new);
+        register("overlay", OverlayView::new);
+        register("gradient", Gradient::new);
+        register("zoomdragview", ZoomDragView::new);
+        register("treeview", TreeView::new);
     }
 
-    private Loader()
+    /** The maximum number of cached documents */
+    private static final int CACHE_CAP = 2;
```

Comment:
Probably client config, not sure about the default value tho (might be worth increasing it).

Reply:
Whoops, that was just for testing. Somewhere between 10 and 30 probably. I'm just trying to think of how many GUIs people access in an hour.